servicex/servicex_adapter.py

```diff
@@ -125,6 +128,37 @@ async def get_datasets(self, did_finder=None) -> List[CachedDataset]:
             datasets = [CachedDataset(**d) for d in r.json()['datasets']]
             return datasets
 
+    async def get_dataset(self, dataset_id=None) -> CachedDataset:
+        headers = await self._get_authorization()
+
+        with httpx.Client() as client:
+            path_template = '/servicex/datasets/{dataset_id}'
+            url = self.url + path_template.format(dataset_id=dataset_id)
+            r = client.get(headers=headers,
+                           url=url)
+
+            if r.status_code == 403:
+                raise AuthorizationError(
+                    f"Not authorized to access serviceX at {self.url}")
+
+            dataset = CachedDataset(**r.json())
+            return dataset
+
+    async def delete_dataset(self, dataset_id=None) -> bool:
+        headers = await self._get_authorization()
+
+        with httpx.Client() as client:
+            path_template = '/servicex/datasets/{dataset_id}'
+            url = self.url + path_template.format(dataset_id=dataset_id)
+            r = client.delete(headers=headers,
+                              url=url)
+
+            if r.status_code == 403:
+                raise AuthorizationError(
+                    f"Not authorized to access serviceX at {self.url}")
+
+            return r.status_code == 200
```

Comment:
If status code is not 200, do you want to give the user some indication of what happened?

Reply:
Used the new _extract_message function to get more extensive error reporting. This required moving to the aiohttp library like the other calls